safekeeper/src/state.rs

```diff
@@ -0,0 +1,197 @@
+//! Defines per timeline data stored persistently (SafeKeeperPersistentState)
+//! and its wrapper with in memory layer (SafekeeperState).
+
+use std::ops::Deref;
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use utils::{
+    id::{NodeId, TenantId, TenantTimelineId, TimelineId},
+    lsn::Lsn,
+};
+
+use crate::{
+    control_file,
+    safekeeper::{AcceptorState, PersistedPeerInfo, PgUuid, ServerInfo, TermHistory},
+};
+
+/// Persistent information stored on safekeeper node
+/// On disk data is prefixed by magic and format version and followed by checksum.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct SafeKeeperPersistentState {
```

Comment:
Let's rename it to just `PersistentState`

Reply:
Maybe `TimelinePersistentState`? A bit too generic otherwise.

Reply:
Works for me, but I'm more a fan of a short names, like `state::Persistent` :)

Reply:
There is also SafekeeperState, and making it just State is too much, so renamed all to Timeline*